flow/directives.py

```diff
@@ -144,7 +144,9 @@ def _add_directive(self, directive):
                 f"Expected a _Directive object. Received {type(directive)}."
             )
         if directive._name in self._directive_definitions:
-            raise ValueError(f"Cannot redefine directive name {directive._name}.")
+            raise FlowProjectDefinitionError(
+                f"Cannot redefine directive name {directive._name}."
+            )
```

Comment:
This method is private (underscored) and not meant to be called by the user. This is more to ensure that we don't violate this when we define an environment's default directives. We can revert this change.